File: Linear_Search/Ruby/linear_search_test.rb
require './linear_search'

describe "linear_search test" do
  it "empty array" do
    linear_search([], 1).should eq(-1)
  end

  it "Search for an element in array" do
    linear_search([1,3,4,10,32], 10).should eq(3)
  end
end

File: Linear_Search/Scala/aayushKumarJarvis/LinearSearchTest.scala
import org.scalatest.{Matchers, FunSuite}

class LinearSearchScalaTest extends FunSuite with Matchers {

  test("Linear Search should search") {
    val objectForLinearSearch = new LinearSearchScala

    objectForLinearSearch.linearSearch(List(1,2,3,4,5),5) should be(true)
    objectForLinearSearch.linearSearch(List(1,2,3,4,5),7) should be(false)
  }
}


